documentation/remote-host-object.md

```diff
@@ -0,0 +1,27 @@
+# Support for remote host object
+
+Remote host object need to be registered in [Running Object Table(ROT)](https://docs.microsoft.com/en-us/windows/desktop/api/objidl/nn-objidl-irunningobjecttable) before hand. And msbuild will call [IRunningObjectTable::GetObject](https://docs.microsoft.com/en-us/windows/desktop/api/objidl/nf-objidl-irunningobjecttable-getobject) to get the host object by monikerName which is passed in method `void RegisterHostObject(string projectFile, string targetName, string taskName, string monikerName)`.
+
+[The registration of interfaces](https://docs.microsoft.com/en-us/dotnet/framework/interop/how-to-register-primary-interop-assemblies) is the only thing interop with COM that need extra care. Registration could be global with register or without. Other interface needed is registered globally by Visual Studio during installation. Ony `ITaskHost` does not use global registration. On VS side, the proxy is written. But on MsBuild side it was not. That result the change in msbuild.exe.manifest file and the change to generate tlb file for ITaskHost.
```

Comment:
I don't understand this. Is this a history of design decisions, or an explanation of relevant facts for the "before" situation? Both are super helpful!

Reply:
My bad writing :( this are all facts. There is no other alternative solutions. Let me see if I can write it better